frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/cim/AIClusterProgress.tsx

```diff
@@ -36,13 +41,35 @@ const fetchEvents = async (url: string) => {
 
 const AIClusterProgress: React.FC = () => {
     const { clusterDeployment, agentClusterInstall, agents } = useContext(ClusterContext)
-    const clusterAgents = agents
-        ? agents.filter(
-              (a) =>
-                  a.spec.clusterDeploymentName?.name === clusterDeployment?.metadata.name &&
-                  a.spec.clusterDeploymentName?.namespace === clusterDeployment?.metadata.namespace
-          )
-        : []
+
+    const [clusterAgents, cluster] = useMemo(() => {
+        const clusterAgents = agents
+            ? agents.filter(
+                  (a) =>
+                      a.spec.clusterDeploymentName?.name === clusterDeployment?.metadata.name &&
+                      a.spec.clusterDeploymentName?.namespace === clusterDeployment?.metadata.namespace
+              )
+            : []
+
+        const cluster = getAICluster({ clusterDeployment, agentClusterInstall, agents: clusterAgents })
+
+        return [clusterAgents, cluster]
+    }, [clusterDeployment, agentClusterInstall, agents])
+
+    const onFetchEvents: CIM.EventListFetchProps['onFetchEvents'] = async (params, onSuccess, onError) => {
```

Comment:
We could have kept getOnFetchEvents (and maybe call it 'getFetchEventsHandler') in ui-lib and just pass `fetchEvents` to it. The API of this function is specific to ui-lib anyway.

getFetchEventsHandler(fetchEvents, agentClusterInstall)